### PR TITLE
[MOE]move weight transpose to wakeup for RL secnarios

### PR DESCRIPTION
**Pull request by lhp-deep**
### What this PR does / why we need it?
In reinforcement learning scenarios, the current inference applies a transpose operation to the weights. For a cleaner architecture, the weight transpose module was moved to wakeup.
### How was this patch tested?

- vLLM version: v0.11.2
- vLLM main: https://github.com/vllm-project/vllm/commit/v0.11.2

### CODE REVIEW

File: vllm_ascend/worker/worker_v1.py [hunk 1]
```diff
@@ -193,6 +193,26 @@ def wake_up(self, tags: Optional[list[str]] = None) -> None:
...
+                w2_data = torch.nn.Parameter(w2_data, requires_grad=False)
+                setattr(parent_module, param_name, w2_data)
+            elif 'w13_weight' in name and param.shape[1] == hidden_size:
+                parts = name.split('.')
+                param_name = parts[-1]
+                parent_module = model.get_submodule(".".join(parts[:-1]))
+
+                w13_data = param.transpose(1, 2)
+                w13_data = torch.nn.Parameter(w13_data, requires_grad=False)
+                setattr(parent_module, param_name, w13_data)
```
Comment (Ronald1995) [inline vllm_ascend/worker/worker_v1.py:211]: duplicate code, please extract a method.

File: vllm_ascend/worker/worker_v1.py [hunk 2]
```diff
@@ -193,6 +193,26 @@ def wake_up(self, tags: Optional[list[str]] = None) -> None:
         allocator = CaMemAllocator.get_instance()
         allocator.wake_up(tags=tags)
 
+        hidden_size = self.vllm_config.model_config.hf_config.hidden_size
```
Comment (leo-pony) [inline vllm_ascend/worker/worker_v1.py]: Please add a NZ e2e accuracy test case to protect the NZ supporting feature.
